app/controllers/chargeback_report_controller.rb

```diff
@@ -5,71 +5,46 @@ class ChargebackReportController < ApplicationController
   after_action :set_session_data
 
   include Mixins::SavedReportPaging
+  include Mixins::GenericListMixin
   include Mixins::GenericSessionMixin
+  include Mixins::GenericShowMixin
   include Mixins::BreadcrumbsMixin
 
   def self.table_name
-    @table_name ||= "chargeback_report"
+    @table_name ||= "miq_report_result"
   end
 
-  def x_show
-    @explorer = true
+  def self.model
+    MiqReportResult
   end
 
-  def tree_select
-    self.x_active_tree = params[:tree] if params[:tree]
-    self.x_node = params[:id]
-    get_node_info(x_node)
-    replace_right_cell
+  def show_list
+    options = {:named_scope => :with_saved_chargeback_reports, :no_checkboxes => true}
+    process_show_list(options)
+    @title = _("Chargeback Saved Reports")
   end
 
-  def explorer
-    @breadcrumbs = []
-    @explorer    = true
-    build_accordions_and_trees
-
-    @right_cell_text = _("All Saved Chargeback Reports")
-    render :layout => "application" unless request.xml_http_request?
-  end
-
-  def title
-    @title = _("Chargeback Reports")
+  def show
+    @center_toolbar = 'chargeback_report'
+    fetch_saved_report(params[:id])
```

Comment:
This is definitely not the right way, we have a mechanism to pull out reports automatically on other screens. Can we use thath instead?

Reply:
that's being used in https://github.com/ManageIQ/manageiq-ui-classic/pull/7058/files#diff-f722efe6718e27e6fd75d892b6a018bcR38-R39
Report object is fetched and processed further